src/x11/x.c

```diff
@@ -1014,7 +1070,44 @@ void x_setup(void)
         x_win_setup();
         x_cairo_setup();
         x_shortcut_grab(&settings.history_ks);
+}
+
+void x_win_round_corners(void)
+{
+        XWindowAttributes win_attr;
+        XGetWindowAttributes(xctx.dpy, xctx.win, &win_attr);
+
+        int width = win_attr.width + win_attr.border_width;
+        int height = win_attr.height + win_attr.border_width;
+
+        Pixmap mask = XCreatePixmap(xctx.dpy, xctx.win, width, height, 1);
+        XGCValues xgcv;
+
+        GC shape_gc = XCreateGC(xctx.dpy, mask, 0, &xgcv);
+
+        int rad = settings.corner_radius;
+        int dia = 2 * rad;
+
+        XSetForeground(xctx.dpy, shape_gc, 0);
+        XFillRectangle(xctx.dpy, mask, shape_gc, 0, 0, width, height);
+
+        XSetForeground(xctx.dpy, shape_gc, 1);
+
+        XFillArc(xctx.dpy, mask, shape_gc, 0, 0, dia, dia, 0, 23040);
```

Comment:
* `2304` is an ugly way to say `360*64`, it'll be optimized away by the compiler anyway and it's much better for readability to write it out.

* This is the first time I'm looking into the X drawing API so correct me if I'm wrong but is there a reason you're drawing the entire 360 degrees instead of just the ones needed? Is there a downside to it?

Reply:
To the second point: There's no specific reason to draw the entire 360 degrees. I just did it for simplicity. I'm going to test the behaviour with just filling the need arc.